django/db/models/sql/where.py

```diff
@@ -118,6 +118,17 @@ def as_sql(self, compiler, connection):
         None, [] if this node is empty, and raise EmptyResultSet if this
         node can't match anything.
         """
+        if self.negated and all(
+            isinstance(child, WhereNode) and child.negated for child in self.children
+        ):
+            return self.create(
+                [
+                    child.create(child.children, child.connector)
+                    for child in self.children
+                ],
+                self.connector,
+            ).as_sql(compiler, connection)
+
```

Comment:
Commit message has "NOT (NOT A0 ... NOT AN) can be simplified to A0 ... NOT AN"

Isn't de morgan basically flipping and/or and !/(not!).

So "NOT (NOT A0 AND ... NOT AN) <=> (A0 OR ... AN)" 🤔

Reply:
Just an observation: the processing path is slightly different for `all()`'s vacuous truth, ie `self.negated` is true but `self.children` is empty.

I _think_ the output is the same but it's always worth double-checking.